packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -202,6 +199,59 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
         }
         dataStoreConfigurationBuilder.errorHandler(errorHandler)
 
+        var conflictHandler: DataStoreConflictHandler
+        conflictHandler = if ((request["hasConflictHandler"] as? Boolean? == true)) {
+            DataStoreConflictHandler { conflictData,
+                                       onDecision ->
+
+                val modelName = conflictData.local.modelName
+                val args = hashMapOf(
+                        "modelName" to modelName,
+                        "local" to FlutterSerializedModel(conflictData.local as SerializedModel).toMap(),
+                        "remote" to FlutterSerializedModel(conflictData.remote as SerializedModel).toMap()
+                )
+
+                uiThreadHandler.post {
+                    channel.invokeMethod("conflictHandler", args, object : Result {
+                        override fun success(result: Any?) {
+                            val resultMap: Map<String, Any>? = result.safeCastToMap()
+                            when (resultMap?.get("resolutionStrategy") as String) {
+                                "APPLY_REMOTE" -> onDecision.accept(DataStoreConflictHandler.ConflictResolutionDecision.applyRemote())
```

Comment:
Maybe we can reuse the existing enum `DataStoreConflictHandler.ResolutionStrategy` from amplify-android for consistency.

Reply:
switched to using enum :

```
try {
                                val resolutionStrategy: ResolutionStrategy =
                                        ResolutionStrategy.valueOf(resultMap?.get("resolutionStrategy") as String)
                                when (resolutionStrategy) {
```